lib/sidekiq/redis_connection.rb

```diff
@@ -7,10 +7,7 @@ class RedisConnection
     class << self
 
       def create(options={})
-        url = options[:url] || determine_redis_provider
-        if url
-          options[:url] = url
-        end
+        options[:url] ||= determine_redis_provider
```

Comment:
I'm worried this will change the logic slightly.  The only difference is that it will create `:url => nil` if the ENV is not set but I don't know if the Redis client will work the same with it.